Common/Securities/SecurityDefinitionSymbolResolver.cs

```diff
@@ -172,18 +212,32 @@ private Symbol SecurityDefinitionToSymbol(SecurityDefinition securityDefinition,
             // The mapped ticker will be null if the map file is null or there's
             // no entry found for the given trading date.
             var mappedTicker = mapFile?.GetMappedSymbol(tradingDate, null);
-            
+
             // If we're null, then try again; get the last entry of the map file and use
             // it as the Symbol we return to the caller.
             mappedTicker ??= mapFile?
                 .LastOrDefault()?
                 .MappedSymbol;
 
-            return string.IsNullOrWhiteSpace(mappedTicker) 
-                ? null 
+            return string.IsNullOrWhiteSpace(mappedTicker)
+                ? null
                 : new Symbol(securityDefinition.SecurityIdentifier, mappedTicker);
         }
 
+        /// <summary>
+        /// Gets the SecurityDefinition corresponding to the given Lean <see cref="Symbol"/>
+        /// </summary>
+        private SecurityDefinition SymbolToSecurityDefinition(Symbol symbol)
+        {
+            if (symbol == null)
+            {
+                return null;
+            }
+
+            return GetSecurityDefinitions()
+                .FirstOrDefault(x => x.SecurityIdentifier.ToString().Equals(symbol.ID.ToString(), StringComparison.OrdinalIgnoreCase));
```

Comment:
> .ToString().Equals

Believe instead of comparing the string we can directly compare the `SecurityIdentifier` itself?